proxy/logging/LogObject.h

```diff
@@ -283,7 +284,7 @@ class LogObject : public RefCountObj
   int m_min_rolled;            // minimum number of rolled logs to be kept, 0 no limit
   bool m_reopen_after_rolling; // reopen log file after rolling (normally it is just renamed and closed)
 
-  head_p m_log_buffer; // current work buffer
+  ts::Atomic_versioned_ptr m_log_buffer; // current work buffer
```

Comment:
The version for this pointer is not used as a simple wrapping counter.  I am not able to understand how it is used.  When the version is stored in spare bits in a 64-bit pointer, it only has 15 bits of precision (so max value 32767).  I am worried that may not be enough in some cases.  This is an issue, since I want to change to favoring the spare bits approach that only needs a 64-bit CAS versus a 128-bit CAS.

Reply:
This is now less of a problem as I'm switching back to favoring using 128-bit CAS with the version stored separately from the pointer.